src/components/ToastNotifications.tsx

```diff
@@ -0,0 +1,190 @@
+import { ICONS, Icon, Notification } from "@canonical/react-components";
+import { DefaultTitles } from "@canonical/react-components/dist/components/Notification/Notification";
+import {
+  GroupedNotificationCount,
+  ToastNotificationType,
+} from "context/toastNotificationProvider";
+import React, { FC, useEffect, useState } from "react";
+import { createPortal } from "react-dom";
+import classnames from "classnames";
+import { Filters, iconLookup, severityOrder } from "util/notifications";
+
+interface Props {
+  notifications: ToastNotificationType[];
+  onDismiss: (notification: ToastNotificationType) => void;
+  onDismissAll: () => void;
+  onDismissByFilter: (
+    notifications: ToastNotificationType[],
+    filters: Filters,
+  ) => void;
+  groupedCount: GroupedNotificationCount;
+}
+
+const ToastNotifications: FC<Props> = ({
+  notifications,
+  onDismiss,
+  onDismissAll,
+  onDismissByFilter,
+  groupedCount = {},
+}) => {
+  const [expanded, setExpanded] = useState(false);
+  const [filters, setFilters] = useState<Filters>({});
+  const hasFilters = !!Object.keys(filters).length;
+
+  useEffect(() => {
+    if (notifications.length < 2) {
+      setExpanded(false);
+      setFilters({});
+    }
+  }, [notifications.length]);
+
+  const handleFilterSelect = (filter: string) => {
+    setFilters((prev) => {
+      const newFilters = { ...prev };
+      if (!newFilters[filter]) {
+        newFilters[filter] = true;
+      } else {
+        // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+        delete newFilters[filter];
+      }
+      return newFilters;
+    });
+  };
+
+  const handleExpansion = () => {
+    setExpanded((prev) => !prev);
+    if (expanded) {
+      setFilters({});
+    }
+  };
+
+  const handleDismissGroups = () => {
+    if (hasFilters) {
+      onDismissByFilter(notifications, filters);
+      setFilters({});
+      return;
+    }
+
+    onDismissAll();
+  };
+
+  const getSeverityFilters = () => {
+    const filterButtons = severityOrder.map((severity) => {
+      if (groupedCount[severity]) {
+        return (
+          <button
+            key={`notification-filter-${severity}`}
```

Comment:
```suggestion
            key={severity}
```
Can keep the keys simpler.